angr/manager.py

```diff
@@ -238,49 +238,50 @@ def _one_state_step(self, a, successor_func=None, resilience=None, **kwargs):
         try:
             # if we have a hook, use it. If we succeed at using the hook, we will break out of this loop
             # otherwise we execute the else-clause, which does the normal step procedure
-            for hook in self._hooks_step_state:
+            for tech_hook in self._hooks_step_state:
                 # FIXME hack to handle some hooks not expecting strong_reference
-                argspec = inspect.getargspec(hook)
+                argspec = inspect.getargspec(tech_hook)
                 if argspec.keywords is None and "strong_reference" not in argspec.args:
                     del kwargs["strong_reference"]
 
-                out = hook(a, **kwargs)
-                if out is not None:
-                    if isinstance(out, tuple):
+                # The technique hook should return a dictionary (stash: [states]) to be merged
+                tech_states_dict = tech_hook(state, **kwargs)
+                if tech_states_dict is not None:
+                    if isinstance(tech_states_dict, tuple):
                         l.warning('step_state returning a tuple has been deprecated! Please return a dict of stashes instead.')
-                        a, unconst, unsat, p, e = out
```

Comment:
I'm talking about this. This `a` is not a state but a list. Pretty confusing but shouldn't be entrenched further.

Reply:
Check out the next commit. I have renamed state into active, since the variable gets overwritten there.